pkg/cloudprovider/aws/block_store.go

```diff
@@ -180,6 +191,29 @@ func (b *blockStore) CreateSnapshot(volumeID, volumeAZ string, tags map[string]s
 	return *res.SnapshotId, nil
 }
 
+func getTagsForCluster(clusterTags map[string]string, snapshotTags []*ec2.Tag) []*ec2.Tag {
+	var result []*ec2.Tag
+
+	// set current cluster ownership tags
+	for k, v := range clusterTags {
+		result = append(result, ec2Tag(k, v))
+	}
+
+	_, haveAWSClusterNameEnvVar := os.LookupEnv("AWS_CLUSTER_NAME")
+
+	for _, tag := range snapshotTags {
+		if haveAWSClusterNameEnvVar && strings.HasPrefix(*tag.Key, "kubernetes.io/cluster/") {
```

Comment:
if `haveAWSClusterNameEnvVar` is true, we should also skip the `KubernetesCluster` tag